code/game/machinery/shieldgen.dm

```diff
@@ -232,79 +234,67 @@
 
-/obj/machinery/shieldwallgen/xenobiologyaccess		//use in xenobiology containment
+/obj/machinery/power/shieldwallgen/xenobiologyaccess		//use in xenobiology containment
 	name = "xenobiology shield wall generator"
 	desc = "A shield generator meant for use in xenobiology."
 	icon_state = "Shield_Gen"
 	req_access = list(ACCESS_XENOBIOLOGY)
 
-/obj/machinery/shieldwallgen/Destroy()
+/obj/machinery/power/shieldwallgen/Initialize()
+	. = ..()
+	if(anchored)
+		connect_to_network()
+
+/obj/machinery/power/shieldwallgen/Destroy()
 	for(var/d in GLOB.cardinals)
 		cleanup_field(d)
 	return ..()
 
-/obj/machinery/shieldwallgen/proc/power()
-	if(!anchored)
-		power = 0
-		return
-	var/turf/T = get_turf(src)
-
-	var/obj/structure/cable/C = T.get_cable_node()
-	var/datum/powernet/PN
-	if(C)
-		PN = C.powernet //find the powernet of the connected cable
-
-	if(!PN)
-		return
-
-	var/surplus = max(PN.avail - PN.load, 0)
-	var/avail_power = min(rand(50,200), surplus)
-	if(avail_power)
-		power += avail_power
-		PN.load += avail_power //uses powernet power.
+/obj/machinery/power/shieldwallgen/should_have_node()
+	return anchored
 
-/obj/machinery/shieldwallgen/process()
-	power()
-	use_stored_power(50)
-
-/obj/machinery/shieldwallgen/proc/use_stored_power(amount)
-	power = CLAMP(power - amount, 0, maximum_stored_power)
-	update_activity()
+/obj/machinery/power/shieldwallgen/connect_to_network()
+	if(!anchored)
+		return FALSE
+	. = ..()
 
-/obj/machinery/shieldwallgen/proc/update_activity()
+/obj/machinery/power/shieldwallgen/process()
 	if(active)
-		icon_state = "Shield_Gen +a"
+		icon_state = "shield_wall_gen_on"
 		if(active == ACTIVE_SETUPFIELDS)
 			var/fields = 0
 			for(var/d in GLOB.cardinals)
 				if(setup_field(d))
 					fields++
 			if(fields)
 				active = ACTIVE_HASFIELDS
-		if(!power)
+		if(!active_power_usage || surplus() >= active_power_usage)
+			add_load(active_power_usage)
+		else
 			visible_message("<span class='danger'>The [src.name] shuts down due to lack of power!</span>", \
+				"If this message is ever seen, something is wrong.",
 				"<span class='italics'>You hear heavy droning fade out.</span>")
-			icon_state = "Shield_Gen"
+			icon_state = "shield_wall_gen"
 			active = FALSE
 			for(var/d in GLOB.cardinals)
 				cleanup_field(d)
 	else
-		icon_state = "Shield_Gen"
+		icon_state = "shield_wall_gen"
 		for(var/d in GLOB.cardinals)
 			cleanup_field(d)
 
-/obj/machinery/shieldwallgen/proc/setup_field(direction)
+/obj/machinery/power/shieldwallgen/proc/setup_field(direction)
 	if(!direction)
 		return
 
 	var/turf/T = loc
-	var/obj/machinery/shieldwallgen/G
+	var/obj/machinery/power/shieldwallgen/G
 	var/steps = 0
 	var/opposite_direction = turn(direction, 180)
 
 	for(var/i in 1 to shield_range) //checks out to 8 tiles away for another generator
 		T = get_step(T, direction)
-		G = (locate(/obj/machinery/shieldwallgen) in T)
+		G = (locate(/obj/machinery/power/shieldwallgen) in T)
```

Comment:
Can you remove these redundant ()'s while you're touching this.

Reply:
took me 500 hours debugging but its done